File: fitrbackend/src/main/java/com/example/fitrbackend/dto/WorkoutSessionResponse.java
package com.example.fitrbackend.dto;

import java.time.Instant;
import java.util.List;
import lombok.Getter;

@Getter
public class WorkoutSessionResponse {

    private final long id;
    private final long user_id;
    private final long workout_location_id;
    private final String locationName;
    private final Instant startTime;
    private final Instant endTime;
    private final String notes;
    private final String title;
    private final List<WorkoutExerciseResponse> workoutExercises;

    public WorkoutSessionResponse(long id, long user_id, long workout_location_id, String locationName,
            Instant startTime, Instant endTime,
            String notes, String title, List<WorkoutExerciseResponse> exercises) {
        this.id = id;
        this.user_id = user_id;
        this.workout_location_id = workout_location_id;
        this.locationName = locationName;
        this.startTime = startTime;
        this.endTime = endTime;
        this.notes = notes;
        this.title = title;
        this.workoutExercises = exercises;
    }
}

File: fitrfrontend/fitrfrontend/API/WorkoutSessionResponse.swift
//
//  WorkoutSessionResponse.swift
//  fitrfrontend
//
//  Created by Ambrose Blay on 2/17/26.
//

import Foundation

struct WorkoutSessionResponse: Codable, Identifiable {
    let id: Int64
    let userId: Int64
    let workoutLocationId: Int64
    let startTime: Date
    let endTime: Date
    let notes: String?
    let workoutExercises: [WorkoutExerciseResponse]
    
    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case workoutLocationId = "workout_location_id"
        case startTime
        case endTime
        case notes
        case workoutExercises
    }
}
